### PR TITLE
CKeystore/CCrypter: move relevant implementation out of the header

### PR DESCRIPTION
**Pull request by jonasschnelli**

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK (haven't verified move-only)

File: src/keystore.cpp [hunk 1]
```diff
@@ -36,6 +36,39 @@ bool CBasicKeyStore::AddKeyPubKey(const CKey& key, const CPubKey &pubkey)
...
+    LOCK(cs_KeyStore);
+    std::set<CKeyID> set_address;
+    for (const auto& mi : mapKeys) {
+        set_address.insert(mi.first);
+    }
+    return set_address;
+}
+
+bool CBasicKeyStore::GetKey(const CKeyID &address, CKey &keyOut) const
+    {
```
Comment (MeshCollider) [inline src/keystore.cpp]: Missing opening brace for function, which is why travis is failing

Comment (jonasschnelli) [reply src/keystore.cpp]: Thanks! Fixed.

**Review by dcousens — APPROVED** [review]

File: src/wallet/crypter.cpp [hunk 2]
```diff
@@ -275,6 +299,19 @@ bool CCryptoKeyStore::GetPubKey(const CKeyID &address, CPubKey& vchPubKeyOut) co
     }
 }
 
+std::set<CKeyID> CCryptoKeyStore::GetKeys() const override
```
Comment (jimpo) [inline src/wallet/crypter.cpp]: Remove `override` (Travis failure)

Comment (jonasschnelli) [reply src/wallet/crypter.cpp]: Fixed. Thanks.

**Comment by benma** [comment]
ACK c6c9661574f7b06aa55e54e6b2805c7ac51ff80e

File: src/keystore.cpp [hunk 3]
```diff
@@ -36,6 +36,40 @@ bool CBasicKeyStore::AddKeyPubKey(const CKey& key, const CPubKey &pubkey)
     return true;
 }
 
+bool CBasicKeyStore::HaveKey(const CKeyID &address) const
+{
+    bool result;
```
Comment (TheBlueMatt) [inline src/keystore.cpp]: Feel like fixing this gook in a new commit while you're at it?

Just LOCK(cs_KeyStore); return mapKeys.count(address) > 0; should work just as well (there are a few more of these later down).

**Review by TheBlueMatt — COMMENTED** [review]
Looks good, would be nice to clean up while you're at it.

**Comment by jonasschnelli** [comment]
Fixed the gook-ish code part mentioned by @TheBlueMatt

**Comment by TheBlueMatt** [comment]
Heh, mind doing that for GetKey, CCryptoKeyStore::IsLocked, CCryptoKeyStore::HaveKey, etc as well? No need for scopes just for return statements.

**Comment by jonasschnelli** [comment]
Added a code-style/folding cleanup commit.

**Comment by TheBlueMatt** [comment]
utACK dd9bb253c3a6be29dd2749ca34e4ec7eb6b593bb

**Comment by laanwj** [comment]
Verified moves-ack dd9bb25

**Pull request merged**